From a703dcd3ff8edae1bfe0308ea673ad47124bdd8c Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 13 Feb 2020 01:57:30 +0200
Subject: [PATCH] [SW-8229] goya2: set otp base after esn

We cant use the first 16 bytes of the otp as factory
is using them for ESN (electrical serial number). Start base of otp
at address 0x10.

Change-Id: I8d539025ba3b3b70bbd809b2fcb23570e5aea740
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/spi_flash.c b/common/spi_flash.c
index 9058aec..df96707 100644
--- a/common/spi_flash.c
+++ b/common/spi_flash.c
@@ -111,7 +111,7 @@
 
 void spi_flash_otp_get_revoke(uint32_t *key_revoke)
 {
-	*key_revoke = spi_flash_readl(FLASH_EFUSE_KEYS_REVOKE_OFFSET) &
+	*key_revoke = spi_flash_readl(FLASH_OTP_KEYS_REVOKE_OFFSET) &
 			SECURITY_KEY_REVOKE_MAP;
 
 	spi_flash_invert_dword(key_revoke);
diff --git a/common/spi_flash.h b/common/spi_flash.h
index c36509c..9f10a09 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -26,8 +26,10 @@
 #define SPI_FLASH_BAUDR_DIVISOR		2
 
 #define FLASH_64KB_PADD			65536
+#define FLASH_FW_VERSION_SIZE		16
+#define FLASH_KEY_HASH_SIZE		160
 
-#define FLASH_OTP_OFFSET		0
+#define FLASH_OTP_OFFSET		0x10
 #define FLASH_OTP_DISABLE		0
 #define FLASH_OTP_ENABLE		1
 
@@ -35,9 +37,12 @@
 #define FLASH_OTP_CMD_EXIT		0xC1
 #define FLASH_OTP_START_TRANSFER	0x1
 
-#define FLASH_OTP_KEYS_OFFSET		0
-#define FLASH_OTP_KEYS_REVOKE_OFFSET	156
-#define FLASH_OTP_FW_VERSION_OFFSET	160
+#define FLASH_OTP_FW_VERSION_OFFSET	FLASH_OTP_OFFSET
+#define FLASH_OTP_KEYS_OFFSET		(FLASH_OTP_FW_VERSION_OFFSET + \
+					 FLASH_FW_VERSION_SIZE)
+#define FLASH_OTP_KEYS_REVOKE_OFFSET	(FLASH_EFUSE_KEYS_OFFSET + \
+					 FLASH_KEY_HASH_SIZE)
+
 #define FLASH_REVOKE_BITS_PER_DWORD	4
 
 #define FLASH_ARC_BASE_ADDRESS		FLASH_START_OFFSET
@@ -48,12 +53,11 @@
 						sizeof(uint32_t))
 #define FLASH_IMAGE_SWITCH_OFFSET	(FLASH_ARC_BASE_ADDRESS + 4096)
 
-/* 16 byte */
 #define FLASH_EFUSE_FW_VERSION_OFFSET	0
-/* 160 bytes */
-#define FLASH_EFUSE_KEYS_OFFSET		16
-/* 16 bytes */
-#define FLASH_EFUSE_KEYS_REVOKE_OFFSET	176
+#define FLASH_EFUSE_KEYS_OFFSET		(FLASH_EFUSE_FW_VERSION_OFFSET + \
+					 FLASH_FW_VERSION_SIZE)
+#define FLASH_EFUSE_KEYS_REVOKE_OFFSET	(FLASH_EFUSE_KEYS_OFFSET + \
+					 FLASH_KEY_HASH_SIZE)
 
 void spi_flash_set_byte_swap(uint8_t enable);
 uint32_t spi_flash_readl(uint32_t offset);
